### PR TITLE
Restyle application UI with CSS framework

### PR DESCRIPTION
**Pull request by hiddewie**
### Reason for this PR

At the moment the UI seems to be outdated with the gradients in the menu bar, the small buttons with color shadows, and half-transparent modals with small buttons and no border.

There is heavy use of jQuery UI, which is also old, and better alternatives exist.

Some options of lightweight frameworks would be [Bulma](https://bulma.io/), [Bootstrap](https://getbootstrap.com/) or [Spectre](https://picturepan2.github.io/spectre/index.html). They can usually used without Javascript.

Attached is a PR which rewrites the main UI of the application. I am wondering if such a thing would be accepted. This is just a start, there may be more to be improved (accordion in modals, usage of icons, etc.).

### Changes in this PR

- Addition of Bulma CSS (https://bulma.io/)
- Removal of much custom CSS
- Addition of Font Awesome icons (used in menu bar, custom leaflet controls, dialogs) (https://fontawesome.com/icons?d=gallery)
- Usage of Bulma modals for presenting dialogs to the user. Replaces all usages of jQuery modals/dialogs.
- Usage of Bulma box and form styles for improving content of many modals. Replaces all occurrences of jQuery accordeon.
- The breakpoint of 'mobile' devices has been changed to 1024px width (in line with Bulma breakpoints).
- Improve readability of Help and Settings dialogs by using semantic HTML.

### Other things (future)

- It would be great if JavaScript ES6 could be used (classes, template litersals, etc.).
- It seems that Travis does not report PR test results on PRs.

### Screenshots of this PR

(Improved main application interface)

image

(Improved modals (here the query wizard))

image

(Improved loading modal with indeterminate loading bar)

image

### CODE REVIEW

**Comment by simon04** [comment]
Hi @hiddewie, thanks for starting to work on a UI refreshment. I like the modernized look and feel a lot. Definitively a +1 for continuing from my side. I'd like to hear @tyrasd's opinion on that proposal!

Regarding the codebase and JavaScript features: I don't see any reasons not to do that. We might need to Babel or other means of transpiling to support older browsers.

**Comment by hiddewie** [comment]
Hello @simon04

Thanks for the heads up! I will continue on this PR to make it work fully with the new interface. I would be willing to work on using newer JavaScript as well, but I think for this PR it would be beneficial to keep it contained to just the look-and-feel (no new functionality or code-rewrite). That would make a sensible review at least partially possible.

I will update here if changes are ready for review.

**Comment by hiddewie** [comment]
@simon04 @tyrasd Could you take a look at the current state? I made a list of all the changes in this PR in the description. This should be the scope of the changes as I had them in mind.

Any feedback is welcome (even "revert this part, it's ugly")!

**Comment by simon04** [comment]
@tyrasd, what do you think?

Wow, this is impressive! I like the fresh look and that the reworked dialogs make better use of the monitor size. Here are some remarks/issues I spotted while testing f66e7c6.

- [ ] The dialogs can no longer be closed using `[Esc]`.
- [ ] Might be personal taste, but I find the two different hues of green hard to distinguesh – 
- [ ] image
- [ ] The buttons in the dialogs could also get some icons
- [ ] The save button does no longer work, but and error is logged on the console
- [ ] The settings button does no longer work, but and error is logged on the console
- [ ] In the load dialog, the middle section "Saved Queries (osm.org)" is broken, an error is logged on the console
- [ ] As an idea, the modal-card-head of the dialogs could show the same icon as the button which triggered them.
- [ ] The spacing in the marker popup is confusing – before: 
image after: image
- [ ] Since we're changing/discussing things, I find the icons on the map somewhat small. Should we enlarge them to the same size of the icons in the menu/toolbar? 
image

**Comment by hiddewie** [comment]
Thanks for the review. The errors in the console should not happen, I will investigate them.

I improved the map view and buttons and search bar a bit.

image

**Comment by hiddewie** [comment]
OK, fixed the following issues:

- jQuery button was still used somewhere, so that was broken (this caused the errors in the save and settings dialog). The import is re-added.
- The button colors are either green or white background now
- The dialog buttons have some icons. The cancel/close button is always white without an icon.
- I cannot reproduce the error in the load dialog. I can load my remote OSM saved queries. I can also log out, and if I open the load dialog again, the
- The modal headers also have the same icons of the buttons that causes it. Good suggestion!
- There was indeed an error in the non-authenticated configuration UI code. Fixed that.

**Comment by hiddewie** [comment]
And fixed the popup in the leaflet map.

image

**Comment by simon04** [comment]
Sorry for taking a few days to respond; during the work week I couldn't motivate myself to do open source projects in the evening. I promise to be more responsive in the next few days.

Awesome work! I really like the new style.

I ticket most of the remarks in https://github.com/tyrasd/overpass-turbo/pull/458#issuecomment-614188532. Here are some more:

- [ ] The spacing in the marker popup is still somewhat confusing – see the margin above/below "Coordinates": Screenshot_2020-04-30 overpass turbo
- [ ] Concerning "Saved Queries (osm.org)": How you I login now? (I'm testing in a private window in Firefox 75.0) Screenshot_2020-04-30 overpass turbo(2)

If you're still motivated, a few additional enhancements could be:

- [ ] Export dialog: convert the action links to buttons?
- [ ] Almost all input fields are rectangular (with a minimal border radius), except for the input field in the Wizard and the Geocoder on the map. Should all get the same border radius?

**Comment by hiddewie** [comment]
Hello @simon04 

No hurries :) Thanks for the feedback.

- Indeed, the popups still had some strange margins... I have tried to fix every combintaion now
  
  
    image

    
    image

- The load screen had a bug. The popup for logging in now pops up when the load screen opens and authentication is enabled.

- I found one more popup which had buttons rendered in an invalid way (the relation without nodes popup). This has been fixed.

I will try to convert the action links to buttons.

As for the border radius on the wizard and the geocoder: this was intentional to distinguish a 'search' input from a 'field'/'data' input (also notice the search icon). This can improve the discoverability of search fields, where users can input anything and receive feedback on their input. For forms with 'field'/'data' inputs, their input has a fixed meaning and should be chosen carefully in the context of the form. I can remove the border radius if you prefer that (either way is fine for me).

**Comment by hiddewie** [comment]
New export window (here in greek, to test the translations)

image

File: locales/en.json [hunk 1]
```diff
@@ -87,7 +87,7 @@
   "export.map_view_expl": "bbox, center, etc.",
 
   "export.section.data": "Data",
-  "export.generic_download_copy": "<a class=\"export\" title=\"saves the exported data as a file\">download</a>/<a class=\"copy\" title=\"copies export output to clipboard\">copy</a> as <span class=\"format\"></span>",
+  "export.generic_download_copy": "<div class=\"field-label is-normal\"><span class=\"format\"></span></div><div class=\"field-body\"><span class=\"buttons has-addons\"><a class=\"export button is-small is-link is-outlined\" title=\"saves the exported data as a file\">download</a><a class=\"copy button is-small is-link is-outlined\" title=\"copies export output to clipboard\">copy</a></span></div>",
```
Comment (hiddewie) [inline locales/en.json:90]: All other translations have been updated with a regex

File: package.json [hunk 2]
```diff
@@ -48,6 +48,8 @@
     "webpack-dev-server": "^2.3.0"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-free": "^5.13.0",
+    "bulma": "^0.8.1",
```
Comment (hiddewie) [inline package.json:52]: CSS framework

File: package.json [hunk 3]
```diff
@@ -48,6 +48,8 @@
     "webpack-dev-server": "^2.3.0"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-free": "^5.13.0",
```
Comment (hiddewie) [inline package.json:51]: Icons

File: README.md [hunk 4]
```diff
@@ -34,5 +34,5 @@ Development is done in the _master_ branch, stable releases are marked with git
 ### install & run
 
 1. `npm install`
-2. `npm start` for a Development server listening at localhost:8080
+2. `npm start` for a Development server listening at http://localhost:8080
```
Comment (hiddewie) [inline README.md:37]: Clickable link in readme

File: css/map.css [hunk 5]
```diff
@@ -15,22 +15,6 @@ body {
...
-    50% 50% no-repeat;
-}
-body.loading .modal {
-  display: block;
-}
```
Comment (hiddewie) [inline css/map.css:33]: Much custom CSS has been removed because it is now part of the CSS framework

File: index.html [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html>
+<html lang="en">
```
Comment (hiddewie) [inline index.html:2]: This attribute is required for modern webpages. Maybe it should be updated once the page has rendered with a certain language

File: js/ide.js [hunk 7]
```diff
@@ -2,12 +2,10 @@
 import $ from "jquery";
 import _ from "lodash";
 import jQuery from "jquery";
-import "jquery-ui/ui/effects/effect-shake";
 import "jquery-ui/ui/widgets/autocomplete";
-import "jquery-ui/ui/widgets/accordion";
-import "jquery-ui/ui/widgets/dialog";
```
Comment (hiddewie) [inline js/ide.js:8]: Many unneeded imports have been removed

**Comment by simon04** [comment]
I'm fine with the rounded search fields, in particular if the change is intended for a better discoverability.

- [ ] Regarding osm.org login: Previously, the osm.org auth page has been opened when expanding the "Saved Queries (osm.org)" section. Now, the osm.org auth page is opened immediately when clicking. A user might not know why this is needed and might be confused why the login dialog is not here. Only working with locally saved queries is a bit tedious now since the osm.org auth page is launched on every click to login. Can we restore the previous behaviour? Maybe having a button in the "Saved Queries (osm.org)" section "Authorize overpass turobo to load/save queries from osm.org" (a better wording is welcome).
- [ ] The English export dialog now has a missing translation: 
Screenshot_2020-05-10 overpass turbo
- [ ] On mobile, the search field on the map is overflowing:
Screen Shot 2020-05-10 at 13 42 34

**Comment by hiddewie** [comment]
Yes, 10d49fc fixed the missing translation. This was a space error that should not have been committed.

Also I wanted to improve the export dialog links even more, but the content is heavily set in the translations. For some of the image links I might break some of the translations in foreign languages.

**Comment by hiddewie** [comment]
The load OSM queries has a button now with 'Load', that pops up the Oauth popup when clicked. If the user is already authenticated, the saved queries are loaded automatically.

**Comment by hiddewie** [comment]
And the search bar width has also been fixed for mobile.

**Comment by simon04** [comment]
Awesome, this is looking great. I think this PR is ready to be merged, right?

**Comment by hiddewie** [comment]
Yes, I just did a full (self-)review of the diff in its current state, and this all looks as intended!

**Comment by simon04** [comment]
Hi @tyrasd, this wonderful feature is finished. Can you please deploy the current master branch to overpass-turbo.eu? Thanks!

**Comment by simon04** [comment]
@tyrasd, can you please deploy a new version of overpass-turbo to overpass-turbo.eu? Thanks!

**Comment by simon04** [comment]
For now it's live at https://tyrasd.github.io/overpass-turbo/

**Comment by simon04** [comment]
@tyrasd, ping, please deploy! :rocket:

**Pull request merged**